Engineering/SourceControl.md

```diff
@@ -30,4 +30,6 @@ Consistency is crucial, therefore the team has to agree on their approach before
 
 ## Q&A
 ### Q: I just committed a secret - how can I remove it from the git history?
-A: use `git reset` to remove them from the commit history. If you already pushed your changes to a public branch, ensure that all contributors that pulled these changes will pull the now updated branch. See also GitHub's [recommendation how to removed sensitive data from a repo](https://help.github.com/articles/removing-sensitive-data-from-a-repository/). The [BFG Repo-Cleaner](https://rtyley.github.io/bfg-repo-cleaner/) works well. 
+A: use `git reset` to remove them from the commit history. If you already pushed your changes to a public branch, ensure that all contributors that pulled these changes will pull the now updated branch.
+
+  If you have published these changes to a public repo like GitHub, you will frist need to change the secrets. Then follow the instrucions to [remove sensitive data fom a reposlity](https://help.github.com/articles/removing-sensitive-data-from-a-repository/). This can be accoplished through the `git filter-brach` command or using the [BFG Repo-Cleaner](https://rtyley.github.io/bfg-repo-cleaner/) and both options are covered in the instructions.
```

Comment:
just a small typing: [remove sensitive data fom a reposlity] should be changed to [remove sensitive data from a repository]